Issue with linux-mingw docker image

This weeks docker image won't link. I initially thought it was related to SDL, but that doesn't explain the qMain  not being found as well.

I tried fumbling around for a command where I could look at what symbols _are_ being exported, but I don't know where to start

I'm not sure how long it's around for, but at the moment last weeks image can be pulled via
`docker pull yuzuemu/build-environments@sha256:ebe5e43ea2d428dc8c0dabf021aaca9692141fa2bbf022f771b3a62293a25298`

Unrelated, but this ended up being commented out
https://github.com/yuzu-emu/build-environments/blob/50e2946e7dfcb90406f5a79049f4947613018882/linux-mingw/Dockerfile#L57

Maybe something like this?
```docker
RUN echo -e "\nyuzu ALL=(ALL:ALL) NOPASSWD: ALL" >> /etc/sudoers
```
Edit: the sudo access is removed at the end, under cleanup. lat9nq pointed that out to me.


```
[yuzu@2ed1927fae64 build]$ ninja yuzu-cmd
[1/1] Linking CXX executable bin/yuzu-cmd.exe
FAILED: bin/yuzu-cmd.exe
<snipped out>
ld.lld: error: undefined symbol: SDL_main
>>> referenced by libSDL2main.a(SDL_windows_main.o):(.text)
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.


[yuzu@2ed1927fae64 build]$ ninja yuzu
[1/1] Linking CXX executable bin/yuzu.exe
FAILED: bin/yuzu.exe
<snipped out>
ld.lld: error: undefined symbol: qMain(int, char**)
>>> referenced by libqt5main.a(qtmain_win.o):(.text)
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```


`pacman -Q` generates a list of installed packages 

This week
```
> binutils 2.39-3
> gcc 12.2.0-1
> gcc-libs 12.2.0-1
> glibc 2.36-3
> iana-etc 20220715-1
> libtasn1 4.19.0-1
> libtool 2.4.7-5
> mingw-w64-fmt 9.1.0-1
> mingw-w64-sdl2 2.24.0-1
> python-appdirs 1.4.4-7
> python-jaraco.context 4.1.2-1
> python-setuptools 1:62.3.0-1
> python-six 1.16.0-6
> python-urllib3 1.26.12-1
```

Last week
```
< binutils 2.39-2
< gcc 12.1.1-4
< gcc-libs 12.1.1-4
< glibc 2.36-2
< iana-etc 20220603-1
< libtasn1 4.18.0-1
< libtool 2.4.7-3
< mingw-w64-fmt 9.0.0-1
< mingw-w64-sdl2 2.0.22-1
< python-appdirs 1.4.4-6
< python-jaraco.context 4.1.1-2
< python-setuptools 1:62.1.0-1
< python-six 1.16.0-5
< python-urllib3 1.26.11-1
```

--- a/linux-mingw/Dockerfile
+++ b/linux-mingw/Dockerfile
@@ -38,7 +38,6 @@ RUN useradd -m -u 1027 -s /bin/bash yuzu && mkdir -p /tmp/pkgs && \
     mingw-w64-qt5-multimedia \
     mingw-w64-qt5-tools \
     mingw-w64-qt5-winextras \
-    mingw-w64-sdl2 \
     mingw-w64-tools \
     mingw-w64-winpthreads \
     mingw-w64-zlib \
@@ -51,8 +50,6 @@ RUN useradd -m -u 1027 -s /bin/bash yuzu && mkdir -p /tmp/pkgs && \
 RUN pip3 install pefile
 # Compatibility with the old Ubuntu MingW image
 RUN ln -s /usr/x86_64-w64-mingw32/lib/qt /usr/x86_64-w64-mingw32/lib/qt5
-# Get rid of -mwindows from SDL2 CMake configs
-RUN sed -i 's/-mwindows//g' /usr/x86_64-w64-mingw32/lib/cmake/SDL2/sdl2-config.cmake
 # Give yuzu user sudo access for AUR usage
 RUN echo "yuzu ALL=(ALL:ALL) NOPASSWD: ALL" >> /etc/sudoers
 
@@ -64,6 +61,17 @@ USER 1027
 
 # Build/install small AUR packages
 RUN \
+# mingw-w64-sdl2
+    cd && \
+    git clone https://aur.archlinux.org/mingw-w64-sdl2.git && \
+    cd mingw-w64-sdl2 && \
+    # Revert to SDL 2.0.22
+    git checkout bd37953 && \
+    # Only build for x86_64
+    sed -i "s/'i686-w64-mingw32'//g" PKGBUILD && \
+    makepkg -si --noconfirm --noprogressbar && \
+    # Get rid of -mwindows from SDL2 CMake configs
+    sudo sed -i 's/-mwindows//g' /usr/x86_64-w64-mingw32/lib/cmake/SDL2/sdl2-config.cmake && \
 # mingw-w64-clang -- MinGW wrapper for host clang
     cd && \
     git clone https://aur.archlinux.org/mingw-w64-wclang-git.git && \

